common/openapi.yml

```diff
@@ -114,6 +158,115 @@ components:
             $ref: '#/components/schemas/ErrorReport'
 
   schemas:
+    RunSetRequest:
+      type: object
+      properties:
+        workflow_url:
+          type: string
+          description: URL to the workflow script to run
+          example: 'http://www.example.com/example.wdl'
+        workflow_param_definitions:
+          type: array
+          items:
+            $ref: '#/components/schemas/WorkflowParamDefinition'
+          description: Declarative input definition for combination with WDS entities
+        wds_entities:
+          $ref: '#/components/schemas/WdsEntitySet'
+
+    WdsEntitySet:
+      type: object
+      properties:
+        entity_type:
+          type: string
+          example: FOO
+        entity_ids:
+          type: array
+          items:
+            type: string
+          example:
+            - F0011111-1111-1111-1111-111111111111
+      description: The set of WDS entities to use when constructing the inputs.
+
+    WorkflowParamDefinition:
+      type: object
+      properties:
+        parameter_name:
+          type: string
+          example: workflow_input_foo_rating
+        parameter_type:
+          $ref: '#/components/schemas/ParameterValueType'
+        source:
+          $ref: '#/components/schemas/ParameterDefinition'
+
+    ParameterValueType:
+      title: ParameterValueType
+      example: String
+      description: What type of value this parameter expects
+      enum:
+        - String
+        - Int
+        - Boolean
+        - Float
+      type: string
+
+    # This ParameterDefinition holds the 'oneOf' declaration for all the various types that can
+    # be used as parameter definitions.
+    # In addition, in Java it acts as the superclass for the various
+    # ParameterDefinitionXYZ classes below. Because the codegen is imperfect, we have to explicitly
+    # add 'allOf: ParameterDefinition' declarations to those classes to force the codegen to make
+    # them subclasses... but in terms of how the swagger UI works, they seem to make no difference
+    ParameterDefinition:
+      title: ParameterDefinition
+      properties:
+        definitionType:
```

Comment:
Should this be `definition_type` to match other fields?

Reply:
I'll try. I changed this because I was testing whether the problem codegen had was to do with camel vs underscore casing. Assuming it works with an underscore that would indeed be preferable... 🤞

Reply:
Indeed, and alas, the discriminator relies on using camel case for this particular field. I can add a comment to that effect.

Reply:
Resolved by renaming to `type` which doesn't need to worry about casing